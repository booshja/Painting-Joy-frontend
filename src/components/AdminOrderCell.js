// dependencies
import React from "react";
import styled from "styled-components";
// components
import {
    StyledGreenSoloButton,
    StyledOutlineButton,
} from "../containers/styles/adminButtons";
import { StyledP } from "../containers/styles/adminTypography";
import { StyledCell } from "../containers/styles/adminContainers";
import { Link } from "react-router-dom";

const StyledLink = styled(Link)`
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    line-height: 1.2;
    font-weight: 700;
    color: #207070;
    text-decoration: none;
    display: inline;
    margin: 0.5rem 0 0.5rem 1rem;
    max-width: 250px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
`;

const StyledBoldP = styled(StyledP)`
    font-weight: 700;
    display: inline;
`;

const StyledText = styled(StyledP)`
    margin-bottom: 0.5rem;
`;

const StyledOutlineBtn = styled(StyledOutlineButton)`
    font-size: 1rem;
    margin-right: 1rem;

    &:disabled {
        color: #dadada;
        border: 2px solid #dadada;
    }
`;

const StyledGreenSoloBtn = styled(StyledGreenSoloButton)`
    &:disabled {
        background-color: #dadada;
    }
`;

const StyledA = styled.a`
    font-family: "News Cycle", sans-serif;
    font-size: 1.2rem;
    font-weight: 700;
    color: #fb9487;
    text-decoration: none;
    margin: 0.5rem 0 1rem 0;
`;

const StyledButtonContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    align-self: center;
`;

const AdminOrderCell = ({ data, handleMarkShipped, handleMarkComplete }) => {
    // destructure data
    const { id, email, transactionId, status, amount } = data;

    return (
        <StyledCell>
            <StyledBoldP>Order:</StyledBoldP>
            <StyledLink to={`/admin/orders/${id}`}>#{transactionId}</StyledLink>
            <StyledText>Paid: ${amount}</StyledText>
            <StyledText>Status: {status}</StyledText>
            <StyledA href={`mailto:${email}`}>Email Customer</StyledA>
            <StyledButtonContainer>
                <StyledOutlineBtn
                    color="#207a7a"
                    onClick={() => handleMarkShipped(id)}
                    disabled={status === "Confirmed" ? false : true}
                >
                    Mark Shipped
                </StyledOutlineBtn>
                <StyledGreenSoloBtn
                    onClick={() => handleMarkComplete(id)}
                    disabled={status === "Shipped" ? false : true}
                >
                    Mark Complete
                </StyledGreenSoloBtn>
            </StyledButtonContainer>
        </StyledCell>
    );
};

export default AdminOrderCell;
