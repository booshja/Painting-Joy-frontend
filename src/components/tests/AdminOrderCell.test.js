import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AdminOrderCell from "../AdminOrderCell";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <BrowserRouter>
            <AdminOrderCell
                data={{
                    id: 1,
                    email: "johndoe@email.com",
                    transactionId: "transactionid123",
                    status: "Confirmed",
                    amount: "1299.99",
                }}
                handleMarkComplete={null}
                handleMarkShipped={null}
            />
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <AdminOrderCell
                data={{
                    id: 1,
                    email: "johndoe@email.com",
                    transactionId: "transactionid123",
                    status: "Confirmed",
                    amount: "1299.99",
                }}
                handleMarkComplete={null}
                handleMarkShipped={null}
            />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByText, getAllByRole } = render(
        <BrowserRouter>
            <AdminOrderCell
                data={{
                    id: 1,
                    email: "johndoe@email.com",
                    transactionId: "transactionid123",
                    status: "Confirmed",
                    amount: "1299.99",
                }}
                handleMarkComplete={null}
                handleMarkShipped={null}
            />
        </BrowserRouter>
    );

    getByText("Order:");
    getByText("#transactionid123");
    getByText("Paid: $1299.99");
    getByText("Status: Confirmed");
    getByText("Email Customer");
    const buttons = getAllByRole("button");

    expect(buttons[0]).toHaveTextContent("Mark Shipped");
    expect(buttons[0]).not.toBeDisabled();
    expect(buttons[1]).toHaveTextContent("Mark Complete");
    expect(buttons[1]).toBeDisabled();
});
