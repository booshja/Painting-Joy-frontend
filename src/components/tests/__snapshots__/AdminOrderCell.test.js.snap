// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c9 {
  background-color: #207070;
  font-family: "News Cycle",sans-serif;
  font-weight: 700;
  font-size: 1rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 5px 10px;
  border: none;
  border-radius: 4px;
  color: #ffffff;
}

.c9:hover:not(:disabled) {
  color: #207070;
  background-color: #ffffff;
  border: 2px solid #207070;
  padding: 3px 8px;
  cursor: pointer;
}

.c9:disabled {
  background-color: #dadada;
}

.c7 {
  border-radius: 4px;
  background-color: #ffffff;
  font-family: "News Cycle",sans-serif;
  font-weight: 700;
  font-size: 1.2rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 2.5px 5px;
  border: 2px solid #207a7a;
  color: #207a7a;
}

.c7:hover:not(:disabled) {
  color: #ffffff;
  background-color: #207a7a;
  cursor: pointer;
}

.c7:disabled {
  color: #dadada;
  border: 2px solid #dadada;
}

.c1 {
  font-size: 1.2rem;
  font-family: "News Cycle",sans-serif;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border: 1px solid #207070;
  width: 100%;
  padding: 1rem;
  background-color: #ffffff;
}

.c3 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.4rem;
  line-height: 1.2;
  font-weight: 700;
  color: #207070;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline;
  margin: 0.5rem 0 0.5rem 1rem;
  max-width: 250px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2 {
  font-weight: 700;
  display: inline;
}

.c4 {
  margin-bottom: 0.5rem;
}

.c8 {
  font-size: 1rem;
  margin-right: 1rem;
}

.c5 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.2rem;
  font-weight: 700;
  color: #fb9487;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0.5rem 0 1rem 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

<div
    class="c0"
  >
    <p
      class="c1 c2"
    >
      Order:
    </p>
    <a
      class="c3"
      href="/admin/orders/1"
    >
      #transactionid123
    </a>
    <p
      class="c1 c4"
    >
      Paid: $1299.99
    </p>
    <p
      class="c1 c4"
    >
      Status: Confirmed
    </p>
    <a
      class="c5"
      href="mailto:johndoe@email.com"
    >
      Email Customer
    </a>
    <div
      class="c6"
    >
      <button
        class="c7 c8"
        color="#207a7a"
      >
        Mark Shipped
      </button>
      <button
        class="c9"
        disabled=""
      >
        Mark Complete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
